rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/component/ComponentController.java

```diff
@@ -261,6 +259,19 @@ public RepositoryLinksResource process(RepositoryLinksResource resource) {
             restControllerHelper.addEmbeddedModerators(halComponent, moderators);
         }
 
+        if (sw360Component.getDefaultVendorId() != null) {
+            Vendor defaultVendor;
+            if (sw360Component.getDefaultVendor() == null
+                    || sw360Component.getDefaultVendor().getId() != sw360Component.getDefaultVendorId()) {
+                defaultVendor = vendorService.getVendorById(sw360Component.getDefaultVendorId());
+            } else {
+                defaultVendor = sw360Component.getDefaultVendor();
+            }
+            addEmbeddedDefaultVendor(halComponent, defaultVendor);
+            sw360Component.setDefaultVendor(null);
```

Comment:
So that the vendor object is not put directly into the component object. We want only the vendorId here and the complete object in the embedded resources.